web/modules/custom/solution/src/Plugin/Validation/Constraint/UniqueSolutionTitleValidator.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Drupal\solution\Plugin\Validation\Constraint;
+
+use Drupal\Component\Utility\Unicode;
+use Drupal\rdf_entity\Entity\Rdf;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates that a field is unique for the given entity type within a bundle.
+ */
+class UniqueSolutionTitleValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($items, Constraint $constraint) {
+    if (!$item = $items->first()) {
+      return;
+    }
+    $field_name = $items->getFieldDefinition()->getName();
+    /** @var \Drupal\rdf_entity\RdfInterface $entity */
+    $entity = $items->getEntity();
+
+    $entity_type_id = $entity->getEntityTypeId();
+    $id_key = $entity->getEntityType()->getKey('id');
+
+    // If the is_version_of field is not empty, check for the same title.
```

Comment:
Can you explain why this is important? I can see in the code that it is checked that the `is_version_of` field is empty. Why is this important? Why can't we check the title if there is something there?